core/src/main/java/org/apache/iceberg/BaseOverwriteFiles.java

```diff
@@ -163,6 +163,12 @@ protected void validate(TableMetadata base, Snapshot parent) {
     }
   }
 
+  @Override
+  protected boolean useReferencedManifestsWhenFiltering(Snapshot parent) {
+    return parent == null
+        || (startingSnapshotId != null && startingSnapshotId == parent.snapshotId());
```

Comment:
Why not make this the default logic?

Reply:
Not every implementation of `MergingSnapshotProducer` has a concept of a `startingSnapshot` to begin validation from, and overwrite and row delta are the 2 cases where we do have that. Although I will double check to make sure we're not being needlessly restrictive in the default implementation which just returns `false` for this.

Reply:
I'd say we should be very careful with enabling this. We can only do this if the set of manifests that was used for reading didn't change and the operation is compatible. For instance, cherry-picks are not safe.